### PR TITLE
IDEMPIERE-5527 Implement asynchronous rendering of dashboard gadget panel

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5527

- implement the same when pushing the Refresh button

## Checklist:

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java [hunk 1]
```diff
@@ -95,8 +99,20 @@ public void onEvent(Event event) throws Exception {
 		String eventName = event.getName();
 
 		if (eventName.equals(Events.ON_CLICK)) {
-			statusPanel.refresh();
-			statusPanel.updateUI();
+    		BusyDialog busyDialog = new BusyDialog();
+            busyDialog.setShadow(false);
+            getParent().insertBefore(busyDialog, getParent().getFirstChild());
+			ServerPushTemplate template = new ServerPushTemplate(AEnv.getDesktop());
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java:105]: we are in event listener thread, ok to just use "getDesktop()" here

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java [hunk 2]
```diff
@@ -95,8 +99,20 @@ public void onEvent(Event event) throws Exception {
...
+    		BusyDialog busyDialog = new BusyDialog();
+            busyDialog.setShadow(false);
+            getParent().insertBefore(busyDialog, getParent().getFirstChild());
+			ServerPushTemplate template = new ServerPushTemplate(AEnv.getDesktop());
+    		ContextRunnable cr = new ContextRunnable() {
+    			@Override
+				protected void doRun() {
+    				refresh(template);
+    				template.execute(() -> {
+    					busyDialog.detach();
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java:111]: better to use template.executeAsyn instead

Comment (CarlosRuiz-globalqss) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java:111]: I copied that line from DashboardRunnable:146

Must I change it there too?

Comment (hengsin) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java:111]: ah yes, should change that too.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java (hunk 1 above)
Comment (CarlosRuiz-globalqss) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java]: ready

**Review by hengsin — APPROVED** [review]
looks good

**Review by hengsin — APPROVED** [review]

**Pull request merged**